pkg/roachprod/vm/aws/aws.go

```diff
@@ -912,8 +936,41 @@ func (p *Provider) listRegions(
 				return nil
 			}
 			mux.Lock()
+			defer mux.Unlock()
+			ret = append(ret, vms...)
+			return nil
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return nil, err
+	}
+
+	return ret, nil
+}
+
+// listRegionsFiltered lists VMs in the regions with a filter on instance names.
+// The filter makes it more efficient to list specific VMs than listRegions.
+func (p *Provider) listRegionsFiltered(
```

Comment:
nit: `listRegionsFiltered` seems almost the same as `listRegions` above with the exception of the filter. Perhaps `listRegions` can call `listRegionsFiltered` with `names = nil` to reduce duplication?